### ISSUE TITLE
1.0.9: Issues with Dual Proxy Setups

### ISSUE THREAD

**Issue by JerichoTorrent** [kangarko/ChatControl#2744]
### Information about the issue/bug

Context:
1. Dual proxy setup.
 - External server accessible with `torrentsmp.com`
 - Haproxy protocol enabled server using Minehut's API
 2. VelocityControl 1.0.9 and RedisBungee-Proxy-Velocity-0.13.0-SNAPSHOT-all on both servers
 
 So, cross-network chat, PMs sent over Velocity, etc. all seem to be working as expected EXCEPT for players on the 2nd proxy (Minehut.) For those players, PMs are not sending over Velocity. Strangely, you can't join the global channel in write mode (see image)
 
![Screenshot 2024-07-04 232120](https://github.com/kangarko/ChatControl-Red/assets/147216535/1ffd34e4-73fa-45ee-a45c-dfbdb5cd9e6a)
However, cross-server spy, reading global chat, and *sending* PMs works, but when a player tries to reply they get "player is not online". Which is quite strange to me because I would think if you can't write in global chat you shouldn't be able to read it either, nor send PMs across Velocity.

Tl;dr cross-server chat works but seems to not be compatible with a server that uses Minehut's API. Any ideas?

**Comment by JerichoTorrent** [comment]
OH also I should mention. With Redis integration enabled, chats were sending twice (duplicated.) So I had to turn the intergration off. Is there a way to turn this back on without duplicating chats?

**Comment by kangarko** [comment]
Sorry, I'd need more time to debug. Assuming you are on the latest version of velocitycontrol. The minehut issue is probably a hard incompatibility I am afraid.

Please try this build to see if it solves the duplicated chat :
[ChatControl-Red-10.27.6.zip](https://github.com/user-attachments/files/16107442/ChatControl-Red-10.27.6.zip)

**Comment by JerichoTorrent** [comment]
Looks like that fixed the duplicate chats :D thank you!
